python/sglang/srt/managers/schedule_batch.py

```diff
@@ -390,15 +395,17 @@ def batch_sampling_params(self, vocab_size, int_token_logit_bias):
         self.top_ks = torch.tensor(
             [r.sampling_params.top_k for r in reqs], dtype=torch.int, device=device
         )
-        self.frequency_penalties = torch.tensor(
-            [r.sampling_params.frequency_penalty for r in reqs],
-            dtype=torch.float,
-            device=device,
-        )
-        self.presence_penalties = torch.tensor(
-            [r.sampling_params.presence_penalty for r in reqs],
-            dtype=torch.float,
+
+        self.penalizer_orchestrator = penaltylib.BatchedPenalizerOrchestrator(
```

Comment:
If the user does not **explicitly** want to enable these parameters, is it still necessary to apply them here?

Reply:
`BatchedPenalizerOrchestrator` will not do anything unless there is a `Req` with penalties defined.

`BatchedPenalizerOrchestrator` class, when being created, will check if there is any `Req` in the batch with penalty defined, and do the work only if there is any.

Of course we can wrap this with `if` clause to even prevent this class from being initialized, but I think that would just add complexity to the `schedule_batch.py` code while not giving us meaningful performance upside.

Reply:
make sense

Reply:
Thanks for pointing out a good one. I will mention in the comment about how `BatchedPenalizerOrchestrator` will behave briefly right above this line.